### PR TITLE
[SDKS-6575] Add SHA in split Fetcher

### PR DESCRIPTION
**Pull request by nmayorsplit**

### CODE REVIEW

File: client/src/main/java/io/split/client/LocalhostSplitChangeFetcher.java [hunk 1]
```diff
@@ -42,4 +48,26 @@ public SplitChange fetch(long since, FetchOptions options) {
             throw new IllegalStateException("Problem fetching splitChanges: " + e.getMessage(), e);
         }
     }
+
+    private SplitChange processSplitChange(SplitChange splitChange, long changeNumber) throws NoSuchAlgorithmException, UnsupportedEncodingException {
+        SplitChange splitChangeToProcess = LocalhostSanitizer.sanitization(splitChange);
+        if (splitChangeToProcess.till < changeNumber && splitChange.till != -1) {
```
Comment (sanzmauro) [inline client/src/main/java/io/split/client/LocalhostSplitChangeFetcher.java:54]: should be `splitChangeToProcess.till`

File: client/src/main/java/io/split/client/LocalhostSplitChangeFetcher.java [hunk 2]
```diff
@@ -42,4 +48,26 @@ public SplitChange fetch(long since, FetchOptions options) {
...
+            _log.warn("The till is lower than the change number or different to -1");
+            return null;
+        }
+        String splitJson = splitChange.splits.toString();
+        MessageDigest digest = MessageDigest.getInstance("SHA-1");
+        digest.reset();
+        digest.update(splitJson.getBytes());
+        byte [] currHash = digest.digest();
+        if (Arrays.equals(lastHash, currHash) || splitChangeToProcess.till == -1) {
+            lastHash = currHash;
```
Comment (sanzmauro) [inline client/src/main/java/io/split/client/LocalhostSplitChangeFetcher.java:64]: it's not necessary. please remove it.

File: client/src/main/java/io/split/client/LocalhostSplitChangeFetcher.java [hunk 3]
```diff
@@ -42,4 +48,26 @@ public SplitChange fetch(long since, FetchOptions options) {
...
+            splitChangeToProcess.since = changeNumber;
```
Comment (sanzmauro) [inline client/src/main/java/io/split/client/LocalhostSplitChangeFetcher.java:65]: same

File: client/src/main/java/io/split/client/LocalhostSplitChangeFetcher.java [hunk 4]
```diff
@@ -42,4 +48,26 @@ public SplitChange fetch(long since, FetchOptions options) {
...
+            splitChangeToProcess.till = changeNumber;
+            return splitChangeToProcess;
```
Comment (sanzmauro) [inline client/src/main/java/io/split/client/LocalhostSplitChangeFetcher.java:67]: same

File: client/src/main/java/io/split/client/LocalhostSplitChangeFetcher.java (hunk 1 above)
Comment (nmayorsplit) [reply client/src/main/java/io/split/client/LocalhostSplitChangeFetcher.java]: Updated. Thanks!!

File: client/src/main/java/io/split/client/LocalhostSplitChangeFetcher.java (hunk 2 above)
Comment (nmayorsplit) [reply client/src/main/java/io/split/client/LocalhostSplitChangeFetcher.java]: Removed it.

File: client/src/main/java/io/split/client/LocalhostSplitChangeFetcher.java (hunk 4 above)
Comment (nmayorsplit) [reply client/src/main/java/io/split/client/LocalhostSplitChangeFetcher.java]: Updated.

File: client/src/main/java/io/split/client/LocalhostSplitChangeFetcher.java (hunk 3 above)
Comment (nmayorsplit) [reply client/src/main/java/io/split/client/LocalhostSplitChangeFetcher.java]: Updated

**Review by sanzmauro — APPROVED** [review]

**Pull request merged**